feat: release

joostvanderborg opened the pull request:
- Allows using Yivi logins (behind feature flag)
Details: https://github.com/GemeenteNijmegen/mijn-nijmegen/pull/435
- Allows setting a digid LOA (not set by default)
Details: https://github.com/GemeenteNijmegen/mijn-nijmegen/pull/435
- Dependency updates

Setting the use_yivi flag will:
- Use a different frontend template for login page
- Add a login button for Yivi, using the yivi_scope

joostvanderborg committed:
feat: allow yivi attributes (#430)

The bsn can be found in several fields: either the 'sub' claim
or a yivi-specific field can be used. This addition tries (in order)
to get a valid BSN from:
- sub
- pbdf.gemeente.bsn.bsn
- irma-demo.gemeente.personalData.bsn

fixes #369

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4738173240

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #432 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4749119674

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #434 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

joostvanderborg committed:
feat: Allow providing OpenIDConnect scope

Adds an optional param to getLoginUrl to provide the scope. This is
useful if you want to provide several login urls on the same page using
several scope configurations.

joostvanderborg committed:
feat: Create yivi button style

Allow creating a button with the Yivi logo, useful for login with
Yivi functionality.

joostvanderborg committed:
feat: Allow configuring Yivi / Digid LOA

The login request handler can now be configured to provide a Yivi
login button, and to set a specific Digid LOA.

Since providing a LOA higher than 'laag' for Digid disallows Yivi,
we need separate login urls for Yivi / Digid. You can now:
- Set a digid service level
- Set a flag 'useYivi' to enable Yivi. This will add a second button
  for yivi.

There's a separate template for the +yivi-page. The 'useYivi'-prop
should be considered a temporary feature flag. After launch this can be
simplified to a single template.

joostvanderborg committed:
feat: Pass useYivi/Digid LOA in login function

Pass the values of the following environment variables to configure:
- `USE_YIVI` (can be the string value 'true' or anything else for false)
- `DIGID_LOA` (the value for a service level: `DigiD_Laag` for instance)

Setting `USE_YIVI` to 'true' will show the yivi +digid login template.
Setting any value for `DIGID_LOA` will send this in the scope as
`service:${DIGID_LOA}`.

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4760075253

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #436 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

joostvanderborg committed:
chore: lint

joostvanderborg committed:
feat: Add scope parameters and use in login function

- Adds scope params for digid / yivi, combine with general scope
- Adds param for enabling yivi
- Modify login function to use new params

joostvanderborg committed:
chore: param test + linting

joostvanderborg committed:
Merge branch 'acceptance' into feat/yivi-and-digid-loas

joostvanderborg committed:
Merge pull request #435 from GemeenteNijmegen/feat/yivi-and-digid-loas

feat: yivi and digid loa's

joostvanderborg committed:
fix: Remove duplicate param (#437)

joostvanderborg committed:
feat: Set button width to auto (#438)

So buttons end up in one line.

joostvanderborg committed:
feat: Set better param defaults (#439)

joostvanderborg committed:
Less flakey digid selector in test (#440)

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4769862145

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #441 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4775447761

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #442 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4781181366

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #443 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

marnixdessing commented on src/shared/OpenIDConnect.ts:
```diff
@@ -63,7 +63,14 @@ export class OpenIDConnect {
      */
-  getLoginUrl(state: string): string {
+  getLoginUrl(state: string, scope?: string): string {
+    if (!scope) {
```
Fijn dat dit niet kapot gaat mochten we ze ooit samenvoegen

marnixdessing commented on src/app/auth/handleRequest.ts:
```diff
@@ -64,4 +68,31 @@ async function loggedinUserName(bsn: string, apiClient: ApiClient) {
+ * @returns {BSN|false}
+ */
+export function bsnFromClaims(claims: IdTokenClaims): Bsn|false {
+  const possibleClaims = ['sub', 'pbdf.gemeente.bsn.bsn', 'irma-demo.gemeente.personalData.bsn'];
```
Wat zorgt er voor dat ik in productie niet met demo attributen in kan loggen nu? Alleen de OIDC claim die wordt meegestuurd?

joostvanderborg replied:
En het feit dat de hele Yivi-login nog gefeature-flagged is, maar als je zelf de url opgeeft kan dat natuurlijk wel. En ook met die demo. Wat niet ok is als je daar een 'echt' bsn in weet te krijgen. Ga ik aanpassen, denk dat ook het checken wat de bron van de claim is geen kwaad kan, zodat ook alleen Irma-attributen kunnen als je een Yivi-login doet. Ga ik eerst toevoegen. Goeie!

marnixdessing replied:
Daar een echt BSN in krijgen is niet ingewikkeld omdat je voor het Yivi demo schema zelf attributen in mag vullen.

joostvanderborg replied:
In https://github.com/GemeenteNijmegen/mijn-nijmegen/pull/446 oplossing hiervoor uitgevoerd.

Commit:
chore(deps): upgrade dependencies

Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/GemeenteNijmegen/mijn-nijmegen/actions/runs/4792305604

------

*Automatically created by projen via the "upgrade-acceptance" workflow*

Signed-off-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Merge pull request #447 from GemeenteNijmegen/github-actions/upgrade-acceptance

chore(deps): upgrade dependencies

joostvanderborg committed:
feat: Only use allowed yivi attributes (#446)

Yivi attributes are passed as a separate env variable to login and auth.
Only attributes provided this way can be used as valid claims for
retrieving the BSN, and only if useYivi is true.

marnixdessing reviewed (approved):
Opmerkingen van gister zijn opgelost 👍
